apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -336,6 +336,14 @@ type EnvoyListenerConfig struct {
 	// +optional
 	DisableMergeSlashes *bool `json:"disableMergeSlashes,omitempty"`
 
+	// DisableServerHeaderTransformation signifies Envoy will not modify the Server header.
+	// false: OVERWRITE
+	// true:	PASS_THROUGH
+	//
+	// Contour's default is false.
+	// +optional
+	DisableServerHeaderTransformation *bool `json:"disableServerHeaderTransformation,omitempty"`
```

Comment:
@sunjayBhatia given that this is an enum in Envoy, what do you think about making it the same here, with the default being "OVERWRITE"? My worry is that if we make it a bool, but then someone wants the "APPEND_IF_ABSENT" behavior, then we need to change the field type to accomodate.

Reply:
yep thats a fair point, makes the questions about plumbing a boolean through easier as well

Reply:
@Vishal-Chdhry let's go with the above approach, thanks!